AvailableMetersPanel.c

```diff
@@ -91,6 +92,47 @@ const PanelClass AvailableMetersPanel_class = {
    .eventHandler = AvailableMetersPanel_eventHandler
 };
 
+// Handle (&CPUMeter_class) entries in the AvailableMetersPanel
+static void AvailableMetersPanel_addCPUMeters(Panel* super, const MeterClass* type, const ProcessList* pl) {
+   if (pl->cpuCount > 1) {
+      Panel_add(super, (Object*) ListItem_new("CPU average", 0));
+      for (unsigned int i = 1; i <= pl->cpuCount; i++) {
+         char buffer[50];
+         xSnprintf(buffer, sizeof(buffer), "%s %d", type->uiName, Settings_cpuId(pl->settings, i - 1));
+         Panel_add(super, (Object*) ListItem_new(buffer, i));
+      }
+   } else {
+      Panel_add(super, (Object*) ListItem_new(type->uiName, 1));
+   }
+}
+
+typedef struct {
+   Panel* super;
+   unsigned int id;
+   unsigned int offset;
+} DynamicIterator;
+
+static void AvailableMetersPanel_addDynamicMeter(ATTR_UNUSED ht_key_t key, void* value, void* data) {
+   const DynamicMeter* meter = (const DynamicMeter*)value;
+   DynamicIterator* iter = (DynamicIterator*)data;
+   unsigned int identifier = (iter->offset << 16) | iter->id;
+   const char* label = meter->description? meter->description : meter->name;
+   Panel_add(iter->super, (Object*) ListItem_new(label, identifier));
+   iter->id++;
+}
+
+// Handle (&DynamicMeter_class) entries in the AvailableMetersPanel
+static void AvailableMetersPanel_addDynamicMeters(Panel* super, const ProcessList *pl, unsigned int offset) {
+   DynamicIterator iter = { .super = super, .id = 1, .offset = offset };
+   Hashtable_foreach(pl->dynamicMeters, AvailableMetersPanel_addDynamicMeter, &iter);
```

Comment:
NULL deref on platforms w/o dynamic meters (when `DynamicMeters_new` returns NULL).

Reply:
Hmm - that would require the platform to have a DynamicMeter_class entry in its Platform_meterTypes array though, which would be a monumental botch.  So, I don't think this one warrants extra special case handling.

Reply:
Then this at least should get a proper `assert` (if not here, then at least in the `Hashtable_foreach` to check things are non-NULL …